cs4660/search/searches.py

```diff
@@ -0,0 +1,309 @@
+    returns a list of actions going from the initial node to dest_node
+    """
+    edges=[]
+    distance = {}
+    parent = {}
+    visited = {}
+    distance[initial_node] = 0
+    parent[initial_node] = None
+    queue = QUEUE.PriorityQueue()
+
+    for node in graph.nodes:
+        if node != initial_node:
+            distance[node] = sys.maxsize
+            parent[node] = None   
+        
+        queue.put(PrioritizedNode(int(distance[node]),node))
+
+    while not queue.empty():
+        prioritized_node = queue.get()
+        node = prioritized_node.node
+        visited[node] = True
+        edge_value = 0
+        node_info = None
+        
+        neighbors = graph.neighbors(node)
+        
+        for neighbor in neighbors:
+
+            if isinstance(graph, GRAPH.AdjacencyMatrix):
+                parent_index=-1
+                node_index=-1
+                for i in range(0,len(graph.nodes)):
+                    if graph.nodes[i] == neighbor:
+                        node_index=i
+                    if parent is not None and graph.nodes[i] == node:
+                        parent_index=i
+
+                edge_value = graph.adjacency_matrix[parent_index][node_index]
+                node_info = NodeInfo(neighbor,node, GRAPH.Edge(node,neighbor,int(edge_value)))
+
+            elif isinstance(graph, GRAPH.AdjacencyList):
+
+                for edge in graph.adjacency_list[node]:
+                    if edge.to_node == neighbor:
+                        edge_value = edge.weight
+                        node_info = NodeInfo(neighbor,node, edge)
+
+            elif isinstance(graph, GRAPH.ObjectOriented):
+                
+                for edge in graph.edges:
+                    if edge.from_node == node and edge.to_node == neighbor:
+                        edge_value = edge.weight
+                        node_info = NodeInfo(neighbor,node, edge)
+                        
+
+            neighbor_distance = distance[node] + int(edge_value)
+            if neighbor_distance < distance[neighbor]:
+                distance[neighbor] = neighbor_distance
+                parent[neighbor] = node_info
+
+            if neighbor not in visited:
+                queue.put(PrioritizedNode(int(distance[neighbor]),neighbor))
+
+    cur_node = dest_node
+    while parent[cur_node] is not None:
+        node_info = parent[cur_node]
+        edges.append(node_info.edge)
+        cur_node = node_info.parent
+            
+    edges.reverse()
+    return edges
+
+    pass
+
+def a_star_search(graph, initial_node, dest_node):
+    """
+    A* Search
+    uses graph to do search from the initial_node to dest_node
+    returns a list of actions going from the initial node to dest_node
+    """
+    queue = QUEUE.PriorityQueue()
+    
+    parent = {}
+    distance = {}
+    parent[initial_node] = None
+    distance[initial_node] = 0
+
+    queue.put(PrioritizedNode(int(distance[initial_node]),initial_node))
```

Comment:
I don't know why you would even need to cast this to `int`, it should already return int.